Applications/Resnet/jni/cifar_dataloader.cpp

```diff
@@ -93,13 +118,65 @@ void RandomDataLoader::next(float **input, float **label, bool *last) {
 }
 
 Cifar100DataLoader::Cifar100DataLoader(const std::string &path, int batch_size,
-                                       int splits) {
-  /// NYI!
+                                       int splits) :
+  batch(batch_size),
+  current_iteration(0),
+  file(path, std::ios::binary | std::ios::ate) {
+  constexpr char error_msg[] = "failed to create dataloader, reason: ";
+
+  NNTR_THROW_IF(!file.good(), std::invalid_argument)
+    << error_msg << " Cannot open file";
+
+  auto pos = file.tellg();
+  NNTR_THROW_IF((pos % Cifar100DataLoader::SampleSize != 0),
+                std::invalid_argument)
+    << error_msg << " Given file does not align with the format";
+
+  auto data_size = pos / (Cifar100DataLoader::SampleSize * splits);
```

Comment:
It looks not handle when the splited data size is smaller than batch. It looks like just set the iteration_per_epoch is zero and not run. Is it intended?

Reply:
Afaik, we do yet to have a good way to handle the iteration less than the batch size. 
I'll make this error when there is no batch size, in a new PR :)

Reply:
check drop_remainder property in https://www.tensorflow.org/api_docs/python/tf/data/Dataset.

Reply:
The concern is commented at line 134.